src/graph/graph.c

```diff
@@ -241,11 +250,15 @@ void _MatrixSynchronize(const Graph *g, RG_Matrix rg_matrix) {
 	GrB_Matrix_ncols(&n_cols, m);
 	GrB_Index dims = Graph_RequiredMatrixDim(g);
 
+	bool allow_empty_space = Config_GetNodeCreationBuffer();
+	/* The matrix must be resized if its dimensions are smaller than the required
+	 * or if we don't allow empty space and the matrix dimensions do not match the required precisely. */
+	bool require_resize = ((n_rows < dims || n_cols < dims) ||
+						   (!allow_empty_space && (n_rows != dims || n_cols != dims)));
+
```

Comment:
We should simplify this condition

```
bool require_resize =  (n_rows != dims || n_cols != dims);
```

If we do allocate additional space, I expect a resize to occur once the buffer is fully used and a new buffer is allocated.

Reply:
That condition is not equivalent and will trigger matrix shrinks.